class/practice2.2/main.cpp

```diff
@@ -0,0 +1,125 @@
+#include <iostream>
+#include <fstream>
+#include <vector>
+#include <sstream>
+#include <algorithm>
+
+using namespace std;
+
+struct Pixel {
+    int r, g, b;
+};
+
+bool isValidPixel(const Pixel &pixel) {
+    return (pixel.r >= 0 && pixel.r <= 255) && (pixel.g >= 0 && pixel.g <= 255) && (pixel.b >= 0 && pixel.b <= 255);
+}
+
+bool loadPixels(const string &filename, vector<vector<Pixel>> &pixels, string &errorMessage) {
+    ifstream file(filename);
+    if (!file.is_open()) {
+        errorMessage = "Error: Could not open file " + filename;
+        return false;
+    }
+
+    string line;
+    int lineCount = 0;
+    while (getline(file, line)) {
+        if (line.empty()) continue; 
+        stringstream ss(line);
+        vector<Pixel> row;
+        string pixelData;
+        while (getline(ss, pixelData, ' ')) {
+            Pixel pixel;
+            replace(pixelData.begin(), pixelData.end(), ',', ' ');
+            stringstream pixelStream(pixelData);
+            if (!(pixelStream >> pixel.r >> pixel.g >> pixel.b) || !isValidPixel(pixel)) {
+                errorMessage = "Invalid pixel data: " + pixelData;
```

Comment:
it is better to use some type to describe that errors for example std::error or std::expected or std::optional<std::string>